### DISCUSSION TITLE
Auth Persistence Between User Sessions

### DISCUSSION THREAD

**Discussion by mrvillage** [supabase/supabase#2005] [Questions] [answered]
Does the supabase-js library have any sort of in-built functionality to enable sign in persistence? I.e. When the user closes the tab and reopens it two days later they are still signed in to their account. I saw a mention of a refresh_token in the docs but I'm not sure what exactly it refers to or how to obtain one if it relates to my question. Thanks!

**Comment by easy-street** [comment, marked as answer]
yes, this is the behavior when initializing the JavaScript client with the default options. by default, the refresh token is stored in local storage and used by the client to automatically refresh/recover expired sessions.

notes:
- make sure to upgrade to the latest version of `@supabase/supabase-js` for recent bug fixes & enhancements to this functionality
- test as follows:
  * set `JWT Expiry` to a small number (in seconds) in the project dashboard at Settings -> Auth settings -> Configuration
  * sign in to the application, then close the browser/tab
  * re-open after the expiry time; the session will be recovered automatically and the user will be signed into the app

**Reply by mrvillage** [reply]
Oh awesome! Thanks a lot!!
